### ISSUE TITLE
Keyword Density and German umlauts

### ISSUE THREAD

**Issue by andre-st** [wp-seopress/wp-seopress-public#45]
Hello Benjamin,

an acquaintance with a German-language blog reported to me that the SEOPress keyword density of a post was displayed incorrectly: no content or no target keyword. Both were there. If he used target keywords without umlauts, it worked without any problems. 

He's using:
- Wordpress 5.9.1-de_DE 
- PHP Version	7.4.28
- wp-config.php: define('DB_CHARSET', 'utf8')
- php.ini: default_charset = "utf-8"
- MySQL 5.5.60 charset/collation "utf8mb4_unicode_ci" (phpMyAdmin)
- current Firefox on Windows 11 (I have it here with Chromium 99 on Linux too)
- SEOPress 5.5.1

I looked at the code at the point and found that the raw keyword is compared against a text with encoded umlauts: e.g. "ö" against "\&#246;" (not even "\&ouml;" from _htmlentities()_ by the way).
For now, I made a quick and dirty hack (just added `html_entity_decode`) that works for his blog but may not work for others.

File: wp-content/plugins/wp-seopress/inc/admin/ajax.php:
```php
...
//Keywords density
...
if (preg_match_all('#\b(' . $kw . ')\b#iu', 
          html_entity_decode( 
               stripslashes_deep(
                    strip_tags(
                         wp_filter_nohtml_kses(
                              $seopress_get_the_content )))), ENT_NOQUOTES | ENT_HTML5, 'UTF-8' ), $m ))
...
```
![2022-03-09-130211_1920x1022_scrot](https://user-images.githubusercontent.com/26840833/157438681-0712f502-75bc-4b5a-a3ff-7cfc42826e76.png)

**Comment by wp-seopress** [comment]
Hi,
thank you for your feedback, we have opened an issue on our private Github repository to take care of this.

Hope to solve this with our next version.

**Comment by wp-seopress** [comment]
@andre-st We try to reproduce the issue without success.

May I know which page builder do you use please?

**Comment by andre-st** [comment]
As far as I can see it doesn't use any special page builder other than the "classic editor" plugin.
I deactivated the classic editor as a test so that the new Wordpress standard editor appears, but the error message remains the same. I've attached another screenshot, where only the keyword "Königstuhl" is used in the text, in the title, etc. and in the target keyword.
![screencapture-ruegenmagic-de-blog-wp-admin-post-new-php-2022-03-30-18_35_02](https://user-images.githubusercontent.com/26840833/160887825-c78aaaa2-ef3c-4c7d-843e-08c7f27dc9ba.png)

**Comment by andre-st** [comment]
As far as I can see it doesn't use any special page builder other than the "classic editor" plugin.
I deactivated the classic editor as a test so that the new Wordpress standard editor appears, but the error message remains the same. I've attached another two screenshots, where only the keyword "Königstuhl" and "Konigstuhl" (no umlaut, "o" instead of "ö") is used in the text, in the title, etc. and in the target keyword.
![screencapture-2022-03-30-18_52_35](https://user-images.githubusercontent.com/26840833/160891467-a4493f42-9676-4e5f-82d2-f0c791db49cc.png)
